app/src/main/java/org/wikipedia/readinglist/recommended/RecommendedReadingListInterestsFragment.kt

```diff
@@ -0,0 +1,522 @@
+    }
+
+    companion object {
+        fun newInstance(fromSettings: Boolean = false): RecommendedReadingListInterestsFragment {
+            return RecommendedReadingListInterestsFragment().apply {
+                arguments = Bundle().apply {
+                    putBoolean(RecommendedReadingListOnboardingActivity.EXTRA_FROM_SETTINGS, fromSettings)
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun RecommendedReadingListInterestsScreen(
+    uiState: Resource<RecommendedReadingListInterestsViewModel.UiState>,
+    fromSettings: Boolean = false,
+    wikiErrorClickEvents: WikiErrorClickEvents? = null,
+    onItemClick: (PageTitle) -> Unit = {},
+    onCloseClick: () -> Unit,
+    onNextClick: () -> Unit,
+    onRandomizeClick: () -> Unit = {},
+    onSearchClick: () -> Unit
+) {
+    val listState = rememberLazyStaggeredGridState()
+    val collapseHeight = LocalDensity.current.run { 100.dp.toPx() }
+
+    val showTitle = remember {
+        derivedStateOf {
+            listState.firstVisibleItemIndex > 0 || listState.firstVisibleItemScrollOffset > collapseHeight
+        }
+    }
+
+    Scaffold(
+        modifier = Modifier
+            .background(WikipediaTheme.colors.paperColor),
+        topBar = {
+            TopAppBar(
+                title = {
+                    AnimatedVisibility(
+                        visible = showTitle.value,
+                        enter = fadeIn(animationSpec = tween(300)),
+                        exit = fadeOut(animationSpec = tween(300))
+                    ) {
+                        Text(
+                            text = stringResource(R.string.recommended_reading_list_interest_pick_title),
+                            color = WikipediaTheme.colors.primaryColor,
+                            style = WikipediaTheme.typography.h2
+                        )
+                    }
+                },
+                navigationIcon = {
+                    Icon(
+                        imageVector = if (!fromSettings) Icons.AutoMirrored.Filled.ArrowBack else Icons.Default.Close,
+                        contentDescription = stringResource(if (!fromSettings) R.string.search_back_button_content_description else R.string.table_close),
+                        modifier = Modifier
+                            .size(48.dp)
+                            .clickable(onClick = onCloseClick)
+                            .padding(12.dp),
+                        tint = WikipediaTheme.colors.primaryColor
+                    )
+                },
+                colors = TopAppBarDefaults.topAppBarColors(
+                    containerColor = WikipediaTheme.colors.paperColor,
+                    scrolledContainerColor = WikipediaTheme.colors.paperColor
+                )
+            )
+        },
+        containerColor = WikipediaTheme.colors.paperColor
+    ) { paddingValues ->
+        when (uiState) {
+            is Resource.Loading -> {
```

Comment:
I am not seeing the loading indicator when it is loading.

Reply:
Hmm, I definitely see a loading indicator, especially when it does a network call. Maybe this was because I didn't have this PR up-to-date with the feature branch?